libuv-static.pc.in

```diff
@@ -8,5 +8,5 @@ Version: @PACKAGE_VERSION@
 Description: multi-platform support library with a focus on asynchronous I/O.
 URL: http://libuv.org/
 
-Libs: -L${libdir} -luv_a @LIBS@
+Libs: -L${libdir} -l:libuv.a @LIBS@
```

Comment:
Why is there a colon? It breaks the build for me:
```
[133/228] : && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -fPIC -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  wpinet/CMakeFiles/wpinet_dsclient.dir/examples/dsclient/dsclient.cpp.o -o bin/wpinet_dsclient  lib/libwpinetd.a  lib/libwpiutild.a  /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/debug/lib/libfmtd.a  -l:libuv.a  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/lib/libpthread.tbd && :
FAILED: bin/wpinet_dsclient 
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -fPIC -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  wpinet/CMakeFiles/wpinet_dsclient.dir/examples/dsclient/dsclient.cpp.o -o bin/wpinet_dsclient  lib/libwpinetd.a  lib/libwpiutild.a  /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/debug/lib/libfmtd.a  -l:libuv.a  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/lib/libpthread.tbd && :
ld: library not found for -l:libuv.a
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
and removing it resolved this error.

Reply:
-l: is a different flag from -l and specifies a full file name (instead of just the soname)

Reply:
Is this a gcc thing? Because I have a `libuv.a` on my library search patch, but as written in the error message it is not found.

Reply:
Seems that this is only supported by the gcc linker and not the llvm linker.

Reply:
Can you explain (or point to an article) why `-l:libuv.a` is necessary and `-luv` is not enough? It is because if there are a `libuv.a` and a `libuv.so` in the same folder the linker would pick up `libuv.so`?

Reply:
It is specified in `man ld.bfd`

Reply:
Can we not use `${libdir}/libuv.a`?

Reply:
But nearly all *.pc files I have seen are wrong in this respect.

Reply:
True, I don't really know why *.pc files seem to use `-L${libdir}`, when the logical thing would be `${libdir}/libuv.a`

Reply:
Ah, apparently it is because those particular flags have special meaning to pkg-config beyond their use in ld (c.f. PKG_CONFIG_SYSROOT_DIR)

Reply:
Hm ok. But this would mean that I use a pkg-config file from a lib of my host system to use a lib on my target system instead of using the pkg-config file of the lib for the target system directly (via `PKG_CONFIG_PATH`).

Reply:
I probably patch the lib that uses libuv via pkgconfig to use cmake config files ... :D